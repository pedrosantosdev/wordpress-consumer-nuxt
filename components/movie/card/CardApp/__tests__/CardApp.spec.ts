import { render } from '@testing-library/vue'
import CardAppVue from '@/components/movie/card/CardApp/CardApp.vue'

describe('CardAppVue', () => {
	it('should render properly props else case', () => {
		const wrapper = render(CardAppVue, {
			props: {
				movie: {
					id: 'id',
					title: 'title',
					overview: 'overview',
					image: null,
					inCinemas: 'TBA',
					hasFile: false,
					needSync: false,
				},
			},
		})
		expect(wrapper.html()).toMatchSnapshot()
	})
	it('should render properly props if case', () => {
		const wrapper = render(CardAppVue, {
			props: {
				movie: {
					id: 'id',
					title: 'title',
					overview: 'overview',
					image: 'data:image/gif;base64,R0lGODdhAQABAPAAAP8AAAAAACwAAAAAAQABAAACAkQBADs=',
					inCinemas: '01-01-2022',
					hasFile: true,
					needSync: true,
				},
			},
		})
		expect(wrapper.html()).toMatchSnapshot()
	})
})
