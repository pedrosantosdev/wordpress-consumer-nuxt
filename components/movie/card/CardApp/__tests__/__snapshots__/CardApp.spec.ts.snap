// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardAppVue > should render properly props else case 1`] = `
"<div class=\\"lg:w-96 w-full sm:h-64 md:h-auto flex flex-col sm:flex-row p-2 sm:p-4 rounded-2xl shadow-lg gap-5 select-none dark:bg-gray-800 bg-white dark:bg-gray-800 dark:text-white\\">
  <div class=\\"h-52 sm:h-full lg:w-1/3 w-full\\">
    <!--v-if-->
  </div>
  <div class=\\"flex flex-col lg:w-2/3 w-full flex-1 gap-5 sm:p-2\\">
    <div class=\\"flex flex-1 flex-col gap-3\\">
      <div class=\\"w-full max-h-14 overflow-ellipsis\\">
        <h2>title</h2>
      </div>
      <div class=\\"w-full min-h-3 max-h-20 line-clamp-3\\">overview</div>
    </div>
    <div class=\\"mt-auto flex gap-3\\">
      <div class=\\"w-24 h-8\\">TBA</div>
      <div class=\\"w-10 h-8 text-2xl dark:text-white text-black\\">
        <nuxticon name=\\"times\\"></nuxticon>
      </div>
      <div class=\\"w-14 h-8 ml-auto\\">
        <div data-v-5185cce9=\\"\\" class=\\"relative inline-block w-10 mr-2 align-middle select-none transition duration-200 ease-in\\"><input data-v-5185cce9=\\"\\" id=\\"id-sw-input\\" class=\\"toggle-checkbox absolute block w-6 h-6 rounded-full bg-white border-4 appearance-none cursor-pointer\\" type=\\"checkbox\\" name=\\"toggle\\"><label data-v-5185cce9=\\"\\" for=\\"id-sw-input\\" class=\\"toggle-label block overflow-hidden h-6 rounded-full bg-gray-300 cursor-pointer\\"></label></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`CardAppVue > should render properly props if case 1`] = `
"<div class=\\"lg:w-96 w-full sm:h-64 md:h-auto flex flex-col sm:flex-row p-2 sm:p-4 rounded-2xl shadow-lg gap-5 select-none dark:bg-gray-800 bg-white dark:bg-gray-800 dark:text-white\\">
  <div class=\\"h-52 sm:h-full lg:w-1/3 w-full\\"><img class=\\"max-h-full mx-auto\\" src=\\"data:image/gif;base64,R0lGODdhAQABAPAAAP8AAAAAACwAAAAAAQABAAACAkQBADs=\\" alt=\\"title\\"></div>
  <div class=\\"flex flex-col lg:w-2/3 w-full flex-1 gap-5 sm:p-2\\">
    <div class=\\"flex flex-1 flex-col gap-3\\">
      <div class=\\"w-full max-h-14 overflow-ellipsis\\">
        <h2>title</h2>
      </div>
      <div class=\\"w-full min-h-3 max-h-20 line-clamp-3\\">overview</div>
    </div>
    <div class=\\"mt-auto flex gap-3\\">
      <div class=\\"w-24 h-8\\">1/0/2022</div>
      <div class=\\"w-10 h-8 text-2xl dark:text-white text-black\\">
        <nuxticon name=\\"check\\"></nuxticon>
      </div>
      <div class=\\"w-14 h-8 ml-auto\\">
        <div data-v-5185cce9=\\"\\" class=\\"relative inline-block w-10 mr-2 align-middle select-none transition duration-200 ease-in\\"><input data-v-5185cce9=\\"\\" id=\\"id-sw-input\\" class=\\"toggle-checkbox absolute block w-6 h-6 rounded-full bg-white border-4 appearance-none cursor-pointer\\" type=\\"checkbox\\" name=\\"toggle\\"><label data-v-5185cce9=\\"\\" for=\\"id-sw-input\\" class=\\"toggle-label block overflow-hidden h-6 rounded-full bg-gray-300 cursor-pointer\\"></label></div>
      </div>
    </div>
  </div>
</div>"
`;
